gnd/src/main/java/com/google/android/gnd/persistence/remote/FirestoreStorageManager.java

```diff
@@ -61,34 +55,27 @@ private StorageReference getRootMediaDir() {
    * @param fileName Name of the uploaded media
    */
   private StorageReference createReference(String fileName) {
-    return getRootMediaDir().child(fileName + '-' + getFilenameSuffix());
+    return getRootMediaDir().child(fileName);
   }
 
-  /** Converts current timestamp to a string to be used a suffix for uploading media. */
-  private String getFilenameSuffix() {
-    return dateFormat.format(new Date());
+  public Maybe<Uri> getDownloadUrl(String path) {
+    return Maybe.create(
+        emitter ->
```

Comment:
Please use the util class `RxTask.toMaybe()` instead.

Reply:
This can't be used here as the Uri is fetched asynchronously.